handout/a11y/key-concerns/semantic-markup.md

```diff
@@ -0,0 +1,166 @@
+
+There are a couple of new attributes on different elements as well to give the browser even more information. The `alt` attribute has been added to the image to let the browser know that it's a logo image. There's also an `aria-live` attribute on the `main` element. This attribute is part of larger spec known as [Accessible Rich Internet Applications (WAI-ARIA)](https://www.w3.org/TR/wai-aria/) which we'll go over in detail. This is something that lets screen readers know that the content within the `main` tag will be updated on the client-side after the page has loaded and needs to be watched for updates.
+
+
+## Roles and ARIA
+
+The ARIA spec was created as a way for content authors a way to communicate more information regarding the semantics of their application rather than just details on how to render the content. This allows assistive technology to understand what's going on inside an application and relay that information in a structured and streamlined format for users with disabilities. One of the main concepts in the ARIA spec is the *role*.
+
+A [role](https://www.w3.org/TR/wai-aria/roles) defines what the purpose of an html element is within the context of that document or application. Roles are defined by adding an attribute to an html element ie. `role="main"` or are defined by default depending on the html element. Some examples of roles are [list](https://www.w3.org/TR/wai-aria/roles#list), [button](https://www.w3.org/TR/wai-aria/roles#button) or [navigation](https://www.w3.org/TR/wai-aria/roles#navigation) which are the default roles of `ul`, `button` and `nav` respectively. Sometimes however, you may not want or be able to use the standard html element to represent these objects in your application, for example, you may want to create your own button component with it's own distinct logic. In this case you can make use of the `role` attribute:
+
+```
+@Component({
+  selector: 'ngc2-app',
+  template: `
+    <ngc2-notification-button
+      message="Hello!"
+      label="Greeting"
+      role="button">
+    </ngc2-notification-button>
+  `,
+  styles: [`
+    div {
+      display: flex;
+      width: 80px;
+      height: 80px;
+      justify-content: center;
+      align-items: center;
+      background-color: yellow;
+      border-radius: 40px;
+    }
+
+    div:hover {
+      cursor: pointer;
+    }
+  `]
+})
+export class AppComponent { }
+
+@Component({
+  selector: 'ngc2-notification-button',
+  template: `
+    <div (click)="notify()">
```

Comment:
I'd argue its better within the template itself, since this way it's no longer on the consuming application to properly markup the component. It's function is _always_ to be a button, so it should be identified as such.